### PR TITLE
Add Coach edit topic

### PR DESCRIPTION
**Pull request by shuliakleonid**

### CODE REVIEW

File: src/app/services/store/coach-topic-store.service.ts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  getTopic(id: string) {
+    this.couchHttpService.getTopic(id).subscribe({
+      next: (topic) => {
+        this.topic$.next(topic);
+      },
+    });
+  }
+
+  updateQuestion(topic: CoachTopicUpdate) {
```
Comment (RomPavel) [inline src/app/services/store/coach-topic-store.service.ts:36]: updateTopic/createTopic

File: src/app/services/store/coach-topic-store.service.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+          this.topics$.next(topic);
+        },
+      });
+  }
+
+  getTopic(id: string) {
+    this.couchHttpService.getTopic(id).subscribe({
+      next: (topic) => {
+        this.topic$.next(topic);
+      },
```
Comment (RomPavel) [inline src/app/services/store/coach-topic-store.service.ts:32]: lets reload allTopics after any change

File: src/app/models/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts [hunk 3]
```diff
@@ -1,39 +1,56 @@
...
+    @Inject(MAT_DIALOG_DATA) public data: TopicEditDialogData,
+    private coachEditorTopic: CoachTopicStoreService,
+  ) {
+    if (data.isEdit && data.topicName) {
+      this.topic = { ...data };
+    }
+  }
+
+  levelChangeHandler($event: Level): void {
+    const level = Object.keys(this.languageLevel).find((key) => this.languageLevel[key] === $event);
```
Comment (RomPavel) [inline src/app/models/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts:39]: lets put this into some util, or even make this as part of choose-level component logic

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts [hunk 4]
```diff
@@ -62,6 +96,11 @@ export class CoachProfileEditorComponent implements OnInit {
   }
 
   openTopicModal() {
-    this.dialog.open(TopicAddingEditingDialogComponent, { data: { isEdit: false } });
+    this.dialog.open(TopicAddingEditingDialogComponent, { data: { ...emptyTopic, isEdit: false } });
+  }
+
+  ngOnDestroy(): void {
+    this.coachTopic.topics$.unsubscribe();
```
Comment (RomPavel) [inline src/app/models/coach-profile-editor/coach-profile-editor.component.ts:103]: this is not how unsubscribe should work

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts [hunk 5]
```diff
@@ -1,54 +1,88 @@
...
-      this.tables$ = this.coachEdit.questions$;
+      this.questionsSubscribe();
     } else if (tabChangeEvent.index === 1) {
       this.selectedTab = CoachEditorTabs.audition;
-      this.tables$ = of(MOCK_AUDITION_QUESTIONS as any as CoachQuestion[]);
     } else if (tabChangeEvent.index === 2) {
       this.selectedTab = CoachEditorTabs.writingAndSpeaking;
-      this.tables$ = of(MOCK_WRITING_AND_SPEAKING_QUESTIONS as any as CoachQuestion[]);
+      this.coachTopic.getAllTopic();
+      this.coachTopic.topics$
```
Comment (RomPavel) [inline src/app/models/coach-profile-editor/coach-profile-editor.component.ts:68]: Lets put it into separate function like questionsSubscribe

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts (hunk 4 above)
Comment (Inunor) [reply src/app/models/coach-profile-editor/coach-profile-editor.component.ts:103]: You should create you own topics$ var inside this component and unsubbscribe from it, not from service instance

File: src/app/models/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts (hunk 3 above)
Comment (shuliakleonid) [reply src/app/models/coach-profile-editor/topic-adding-editing-dialog/topic-adding-editing-dialog.component.ts:39]: done

File: src/app/services/store/coach-topic-store.service.ts (hunk 2 above)
Comment (shuliakleonid) [reply src/app/services/store/coach-topic-store.service.ts:32]: done

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts (hunk 4 above)
Comment (shuliakleonid) [reply src/app/models/coach-profile-editor/coach-profile-editor.component.ts:103]: done

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts (hunk 5 above)
Comment (shuliakleonid) [reply src/app/models/coach-profile-editor/coach-profile-editor.component.ts:68]: done

File: src/app/models/coach-profile-editor/coach-profile-editor.component.ts (hunk 4 above)
Comment (shuliakleonid) [reply src/app/models/coach-profile-editor/coach-profile-editor.component.ts:103]: done

**Review by RomPavel — APPROVED** [review]

**Pull request merged**